openhcl/underhill_core/src/dispatch/mod.rs

```diff
@@ -471,7 +474,17 @@ impl LoadedVm {
         if self.isolation.is_isolated() {
             anyhow::bail!("Servicing is not yet supported for isolated VMs");
         }
+
+        // capabilities_flags used to explicitly disable the feature
+        // which is enabled by default.
         let nvme_keepalive = !capabilities_flags.disable_nvme_keepalive();
+        if let Some(m) = self.nvme_manager.as_mut() {
+            // Only override if explicitly disabled from host.
+            if !nvme_keepalive {
+                m.override_nvme_keepalive_flag(nvme_keepalive);
```

Comment:
Instead of storing a flag in the `NvmeManager`, let's just take this as a parameter to `shutdown`.

And instead of having `save` fail if `nvme_keepalive` is false, just don't call `save()` below when you don't want to save.

Reply:
It used to be as a parameter to shutdown but I had to redo it this way to cover some failing cases. Let me get back to this with details once I review.

Reply:
To resolve this, I need to add this flag to both `save()` and `shutdown()` and this approach was rejected during earlier reviews. Let me add it back - see if it is better this time.

Reply:
Restored parameter to shutdown, although keeping `update_servicing_flags` is necessary in processing `Request::Shutdown` to avoid unexpected controller reset in `drop`.